library_generation/utilities.sh

```diff
@@ -273,3 +275,19 @@ get_version_from_versions_txt() {
   version=$(grep "$key:" "${versions}" | cut -d: -f3) # 3rd field is snapshot
   echo "${version}"
 }
+
+detect_OS() {
```

Comment:
This function is irrelevant to this PR.

Reply:
My bad, I missed it in the PR separation. I'm removing it